Client/Frontend/Browser/TabManager.swift

```diff
@@ -64,7 +64,7 @@ class TabManager: NSObject {
         }
     }
 
-    fileprivate(set) var tabs = [Tab]()
+    fileprivate(set) var allTabs = [Tab]()
```

Comment:
This is kind of what I'm hoping to get away from. The only tabs that should be in-memory are the tabs for the current mode. If we go to private mode, the "tabs" array should be cleared out and filled with solely private tabs.

Going to normal mode, should dump all existing tabs, and switch to normal tabs. I don't think there should really be a concept of `allTabs`.

Reply:
Considering we no longer keep private tabs at all when the user leaves private mode, we could just re-query the tabs saved in CD when switching back to regular mode.

I think we should merge this anyways for the time being as a part 1 of many changes to TabManager.

We can make the next step removing `allTabs` like this step was removing `tabs(withType:)`

Reply:
Sounds good, will merge, but we still need to refactor this API. Nice intermediate step though!